From d64aab0c6fdc67113f2332a295d244b53bf694d6 Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Wed, 20 Apr 2016 16:28:09 +0200
Subject: [PATCH] hub: admit devices are SS+
Git-Commit: d64aab0c6fdc67113f2332a295d244b53bf694d6
Patch-Mainline: v4.7
References: FATE#319959

If a port can do 10 Gb/s the kernel should say so.
The corresponding check needs to be added.

Signed-off-by: Oliver Neukum <ONeukum@suse.com>

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/usb/core/hub.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 38cc4ba..c2270d8 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -104,6 +104,8 @@ static int usb_reset_and_verify_device(struct usb_device *udev);
 
 static inline char *portspeed(struct usb_hub *hub, int portstatus)
 {
+	if (hub_is_superspeedplus(hub->hdev))
+		return "10.0 Gb/s";
 	if (hub_is_superspeed(hub->hdev))
 		return "5.0 Gb/s";
 	if (portstatus & USB_PORT_STAT_HIGH_SPEED)
-- 
2.1.4

